### ISSUE TITLE
False-positive errors: invalid URL, awesome-git-repo-age, no-defined-references, no-repeat-punctuation, invalid git repository.

### ISSUE THREAD

**Issue by wongyah** [sindresorhus/awesome-lint#191]
# Error message

After running awesome-lint on my repo [Awesome Technical Writing Learning](https://github.com/wongyah/awesome-technical-writing-learning), I got the following error messages:

```
C:\Users\Administrator>npx awesome-lint https://github.com/wongyah/awesome-technical-writing-learning
× Linting

  README.md:6:3ed with exit code 128: git clone --depth 1 -- https://github.com/wongyah/awesome-technical-writing-learni  ×    1:1   Invalid URL                                         remark-lint:double-link
  ×    1:1   Git repository must be at least 30 days old         remark-lint:awesome-git-repo-age
  ×    1:1   Awesome list must reside in a valid git repository  remark-lint:awesome-githubiled to connect to github.co  ×    6:3   Found reference to undefined definition             remark-lint:no-undefined-references
  ×    9:3   Found reference to undefined definition             remark-lint:no-undefined-references
  ×   54:11  Should not repeat "."                               remark-lint:no-repeat-punctuationng
  ×   54:12  Should not repeat "."                               remark-lint:no-repeat-punctuation
  ×  222:3   Found reference to undefined definition             remark-lint:no-undefined-references
  ×  256:39  Should not repeat "."                               remark-lint:no-repeat-punctuation
  ×  256:40  Should not repeat "."                               remark-lint:no-repeat-punctuation

  10 errors
```

# Summary

There are five kinds of false-positive errors above:

## Invalid URL

Repository URL: `https://github.com/wongyah/awesome-technical-writing-learning`.

## Repo age

![my-repo-age](https://github.com/sindresorhus/awesome-lint/assets/38174537/5ba4228c-212c-41ec-b9e7-6a1201001ba1)

## Invalid git repository

em ... 

## Undefined definition

[Github alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) syntax was detected as undefined definitions.

```md
> [!TIP] 
> All resources in this curated list are free, unless otherwise noted.
```

## Repeated punctuation

All the ellipsises (`...`) are detected as repeated dots.

```md
But wait ..., you still need to ...

[Everything You Want to Know about ...](https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-arrays?view=powershell-7.4)
```

**Comment by wongyah** [comment]
I followed @tapyu's suggestion to run awesome-lint again within the local git directory. The invalid URL and invalid repo errors weren't thrown again. But the other error are still there:

```
S E:\Site-and-Repos\wyRepos\awesome-technical-writing-learning> npx awesome-lint https://github.com/wongyah/awesome-technical-writing-learning
✖ Linting

  README.md:6:3
  ✖    1:1   Git repository must be at least 30 days old  remark-lint:awesome-git-repo-age   
  ✖    6:3   Found reference to undefined definition      remark-lint:no-undefined-references
  ✖    9:3   Found reference to undefined definition      remark-lint:no-undefined-references
  ✖  127:3   https://code.visualstudio.com/               remark-lint:double-link
  ✖  161:3   https://code.visualstudio.com/               remark-lint:double-link
  ✖  237:3   Found reference to undefined definition      remark-lint:no-undefined-references
  ✖  271:39  Should not repeat "."                        remark-lint:no-repeat-punctuation  
  ✖  271:40  Should not repeat "."                        remark-lint:no-repeat-punctuation  

  8 errors
```

# Errors

## Repo age

![](https://private-user-images.githubusercontent.com/38174537/322255236-5ba4228c-212c-41ec-b9e7-6a1201001ba1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMzc1NDIsIm5iZiI6MTcxMzMzNzI0MiwicGF0aCI6Ii8zODE3NDUzNy8zMjIyNTUyMzYtNWJhNDIyOGMtMjEyYy00MWVjLWI5ZTctNmExMjAxMDAxYmExLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDA3MDA0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkNjU2YTViMjA2MTZkYjZhNWJlOTY5NDIyNDczOTdmNWQyZjEzN2MxNjI3OWFkZTZhOWZmNGE3NTNkYzgxNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1C2vcGE1cIm7FqkPmstWrZYzTzBgBccy-XaBMldnQOg)

## Undefined definition

[Github alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) syntax was detected as undefined definitions.

```md
> [!TIP] 
> All resources in this curated list are free, unless otherwise noted.
```

## Repeated punctuation

All the ellipsises (...) are detected as repeated dots.

```md
[Everything You Want to Know about ...](https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-arrays?view=powershell-7.4)
```

**Comment by wongyah** [comment]
I replaced `...` with html entity `&hellip;`, then the repeated punctuation error is gone.

I still got 3 kinds of errors:

- Repo age
- Undefined definition caused by [Github alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).
- Double links

### Double links

```
### Markdown

- [Visual Studio Code (VS Code)](https://code.visualstudio.com/)
    - Extension A 
    - Extension B

### reStructuredText

- [Visual Studio Code (VS Code)](https://code.visualstudio.com/)
    - Extension E
    - Extension F
```

If remove one of the VS Code link, one of the lists will be incomplete.

**Comment by batrdn** [comment]
The issue regarding repo age may be due to this [reason](https://github.com/sindresorhus/awesome/pull/3022#issuecomment-2088859642)

**Comment by wongyah** [comment]
@batrdn The `readme.md` file was updated 3 weeks ago, but it was created 4 months ago. It is the first file created in the repository. 

I ever removed several commits using git commands and force push to the GitHub repository. Maybe these operations caused the `awesome-git-repo-age` error.

**Comment by batrdn** [comment]
@wongyah ahh I see, then probably has nothing to do with it.
